pallets/funding/src/tests/4_community.rs

```diff
@@ -1,1224 +1,1394 @@
-
-	let plmc_contribution_funding = MockInstantiator::calculate_contributed_plmc_spent(
-		contributions.clone(),
-		project_details.weighted_average_price.unwrap(),
-	);
-	let plmc_existential_deposits = plmc_contribution_funding.accounts().existential_deposits();
-	inst.mint_plmc_to(plmc_contribution_funding.clone());
-	inst.mint_plmc_to(plmc_existential_deposits.clone());
-
-	let foreign_asset_contribution_funding = MockInstantiator::calculate_contributed_funding_asset_spent(
-		contributions.clone(),
-		project_details.weighted_average_price.unwrap(),
-	);
-	inst.mint_foreign_asset_to(foreign_asset_contribution_funding.clone());
-
-	inst.contribute_for_users(project_id, contributions).unwrap();
-
-	assert_eq!(inst.get_project_details(project_id).remaining_contribution_tokens, 0);
-}
+			let bobs_evaluation_bond =
+				inst.execute(|| Balances::balance_on_hold(&HoldReason::Evaluation(project_id).into(), &BOB));
+			let bobs_slashable_bond = <TestRuntime as Config>::EvaluatorSlash::get() * bobs_evaluation_bond;
+			let bobs_usable_bond = bobs_evaluation_bond - bobs_slashable_bond;
+
+			let plmc_price = <TestRuntime as Config>::PriceProvider::get_price(PLMC_FOREIGN_ID).unwrap();
+			let ct_price = inst.get_project_details(project_id).weighted_average_price.unwrap();
+
+			let usable_usd = plmc_price.saturating_mul_int(bobs_usable_bond);
+			let usable_ct = ct_price.reciprocal().unwrap().saturating_mul_int(usable_usd);
+
+			let slashable_usd = plmc_price.saturating_mul_int(bobs_slashable_bond);
+			let slashable_ct = ct_price.reciprocal().unwrap().saturating_mul_int(slashable_usd);
+
+			// Can't contribute with only the evaluation bond
+			inst.execute(|| {
+				assert_noop!(
+					Pallet::<TestRuntime>::community_contribute(
+						RuntimeOrigin::signed(BOB),
+						get_mock_jwt(BOB, InvestorType::Retail, generate_did_from_account(BOB)),
+						project_id,
+						usable_ct + slashable_ct,
+						1u8.try_into().unwrap(),
+						AcceptedFundingAsset::USDT,
+					),
+					Error::<TestRuntime>::NotEnoughFunds
+				);
+			});
+
+			// Can use half of the usable evaluation bond
+			let contribution_usdt = MockInstantiator::calculate_contributed_funding_asset_spent(
+				vec![(BOB, usable_ct / 2).into()],
+				ct_price,
+			);
+			inst.mint_foreign_asset_to(contribution_usdt.clone());
+			inst.execute(|| {
+				assert_ok!(Pallet::<TestRuntime>::community_contribute(
+					RuntimeOrigin::signed(BOB),
+					get_mock_jwt(BOB, InvestorType::Retail, generate_did_from_account(BOB)),
+					project_id,
+					usable_ct / 2,
+					1u8.try_into().unwrap(),
+					AcceptedFundingAsset::USDT,
+				));
+			});
+
+			// If we mint what we just used, and mint what we cannot use due to being a slash deposit, \
+			// then we can use make a contribution that uses the same amount that we originally bonded on the evaluation
+			let spent_usable_plmc = (BOB, bobs_usable_bond / 2).into();
```

Comment:
I wanted to test using partial and full amount of usable evaluation-bonded PLMC.

In this test we:
1) Try using all of it for a contribution - fails because we need to leave the slash deposit
2) Try partially using half of it - Succeeds
3) Try minting the slashable deposit plus what we just spent so the test 1) can now pass.

I agree point 3 might be confusing, so maybe the 3rd test could be a different account doing the same evaluation, and then using all of the usable deposit without any extra mints?
And then a test 4) that tries to repeat it and fails?